benchar/benchar.py

```diff
@@ -0,0 +1,87 @@
+class benchar:
+    def __init__(self):
+        self.cmp_count = 0
+    
+    def __call__(self, *args, **kwargs):
+        base_str = str(*args, **kwargs)
+        return benchar.count_str(base_str, self)
+
+    def _first_diff(a, b):
+        for i in range(min(len(a), len(b))):
+            if str.__ne__(a[i], b[i]):
+                return i + 1
+        return min(len(a), len(b))
+    
+    class count_str(str):
+        def __new__(cls, base_str, parent_benchar):
+            obj = str.__new__(cls, base_str)
+            obj._parent_benchar = parent_benchar
+            return obj
+
+        def __getitem__(self, key):
+            obj = str.__getitem__(self, key)
+            obj = benchar.count_str(obj, self._parent_benchar)
+            return obj
+        
+        def __add__(self, other):
+            obj = str.__add__(self, other)
+            obj = benchar.count_str(obj, self._parent_benchar)
+            return obj
+        
+        def __iter__(self):
+            obj = str.__iter__(self)
+            obj = benchar.count_str_iterator(obj, self._parent_benchar)
+            return obj
+        
+        def __eq__(self, other):
+            res = str.__eq__(self, other)
+            self._parent_benchar.cmp_count += benchar._first_diff(self, other)
+            return res
```

Comment:
If you don't want to get rid of this implementation, please provide another implementation i.e. a class of strings which would (1) count only its own operations, but (2) will implement also reversed ops (e.g. __radd__), and provide fallbacks e.g. in case 'AAAA' == YourString('AAAB') it would store +4 in its internal field of this instance of YourString

To my knowledge 'AAAA' == YourString('AAAB') should throw NotImplemented, and thus [from the documentation] it follows that:
When NotImplemented is returned, the interpreter will then try the reflected operation on the other type, or some other fallback, depending on the operator. If all attempted operations return NotImplemented, the interpreter will raise an appropriate exception.

Therefore, if a.__eq__(b) fails, there should be an invocation of b.__eq__(a).

PS. I know, when t and w are YourStrings, then we need to sum their counters. Or you can add +1 to all arguments that are YourStrings, that's also a valid solution